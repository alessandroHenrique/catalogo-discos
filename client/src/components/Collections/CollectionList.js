import React from 'react';
import PropTypes from 'prop-types';
import Collection from './Collection';
import uuid from 'uuid';

const CollectionList = ({ collections, deleteCollection, updateCollection }) => (
  <ul className="list-group list-unstyled">
    {
      collections.map((collection) => (
        <Collection
          key={uuid()}
          title={collection.title}
          discs={collection.discs}
          deleteCollection={() => deleteCollection(collection.id)}
          updateCollection={(title) => updateCollection(collection.id, title)}
          collectionId={collection.id}
        />
      ))
    }
  </ul>
);

CollectionList.propTypes = {
  collections: PropTypes.array,
  deleteCollection: PropTypes.func.isRequired,
  updateCollection: PropTypes.func.isRequired
};

export default CollectionList;
