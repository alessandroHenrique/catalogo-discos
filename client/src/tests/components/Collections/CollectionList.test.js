import CollectionList from '../../../components/Collections/CollectionList';
import React from 'react';
import Store from '../../../store/configureStore';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';

const store = Store();

describe('CollectionList', () => {
  it('should render a CollectionList', () => {

    const component = renderer.create(
      <Provider store={store}>
        <CollectionList
          collections={[]}
          deleteCollection={() => {}}
          updateCollection={() => {}}
        />
      </Provider>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  })
})
