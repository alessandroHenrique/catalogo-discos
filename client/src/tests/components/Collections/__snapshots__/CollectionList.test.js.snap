// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionList should render a CollectionList 1`] = `
<ul
  className="list-group list-unstyled"
/>
`;
